SMLHelper/Assets/FishPrefab.cs

```diff
@@ -69,7 +69,11 @@ public sealed override GameObject GetGameObject()
             forces.handleGravity = true;
             forces.underwaterDrag = 1f;
             forces.underwaterGravity = 0;
+#if BELOWZERO
```

Comment:
Having different build results for Subnautica and Below Zero would mean that QModManager would have separate build results for Subnautica and Below Zero after the merge. For now, it is fine, but I will make some modifications when the merge occurs, perhaps by using a custom build action to update the code.

```cs
#if BELOWZERO
    forces.waterDepth = Ocean.GetOceanLevel();
#else 
    forces.waterDepth = Ocean.main.GetOceanLevel();
#endif 
```

```cs
if (Patcher.Game == Game.BelowZero)
{
    // do some reflection shit here
}
else
{
	// do some other reflection shit here
}
```

The only (pseudo-)down-side is that the game will take longer to load by 0.1 seconds.

Anyway, I'm not saying that I will do that, since it's complicated as hell, but a similar custom build event could be used to add _dependency injection to all classes that require it without having to bloat up the repo_.

Reply:
If we merge, it will no longer be possible to have the same QModInstaller exe and the same QModManager dll serving both games.  

SMLHelper will want direct references to the correct Assembly-CSharp.  
And we can't have a project attempting to reference all of them for both games at the same time.

This is another reason why merging SMLHelper into QMM had to be postponed.

Reply:
We load a proxy assembly before Assembly-CSharp.dll loads using Doorstop, we check which game QModManager was installed for, then we load the main assembly with that information already known.

Reply:
I am resolving this discussion because it pertains to QModManager.  
We will continue it there on a new branch when we go back to QMM 3.0